Applications/spire/source/charting/charting_window.cpp

```diff
@@ -142,10 +148,25 @@ ChartingWindow::ChartingWindow(Ref<SecurityInputModel> input_model,
 
 void ChartingWindow::set_model(std::shared_ptr<ChartModel> model) {
   m_model = model;
+  delete m_technicals_panel;
   delete m_chart;
-  m_chart = new ChartView(*m_model, this);
+  delete m_security_widget_container;
+  m_security_widget_container = new QWidget(this);
+  auto container_layout = new QVBoxLayout(m_security_widget_container);
+  container_layout->setContentsMargins({});
+  m_technicals_model = std::make_shared<LocalTechnicalsModel>(Security());
```

Comment:
Never instantiate a model from within a window, ot defeats the whole purpose of using a polymorphivlc model.